lib/resty/etcd/v3.lua

```diff
@@ -688,43 +689,74 @@ local function request_chunk(self, method, path, opts, timeout)
 
 
     local function read_watch()
+        body = nil
 
         while(1) do
-            body, err = res.body_reader()
+            local chunk, error = res.body_reader()
+            if error then
+                return nil, error
+            end
+            if not chunk then
+                break
+            end
+
             if not body then
-                return nil, err
+                body = chunk
+            else
+                body = body .. chunk
```

Comment:
Could we use `table.concat` to avoid creating too much temp str?

Reply:
OK